src/framework/model.cr

```diff
@@ -365,6 +365,8 @@ module Ktistec
               if (o = options[key]).is_a?(typeof(self.{{v}}))
                 @changed << {{v.symbolize}}
                 self.{{v}} = o.as(typeof(self.{{v}}))
+              else
+                Log.warn { "Got invalid type assigned to #{key}" }
```

Comment:
This would have really helped identify the issue with saving, so leaving it in.

Reply:
this needs to go into the initialize methods and the other assign method, but it's probably overdue. i'd recommend you pull the change out of your PR and i'll make all of the necessary changes in commit it myself. keeping a dirty copy of this file around until then is probably a good idea—i've been bitten myself before...